core/src/main/java/org/apache/iceberg/ManifestReader.java

```diff
@@ -217,7 +257,7 @@ private void cacheChanges() {
 
         addCloseable(reader);
 
-        return reader;
+        return inheritableMetadata != null ? CloseableIterable.transform(reader, inheritableMetadata::apply) : reader;
```

Comment:
Nit: I don't think `inheritableMetadata` is ever null now, so we can remove this check and just return the transform.

Reply:
Agree, done.